tests/Feature/Booking/BookingTest.php

```diff
@@ -275,85 +280,142 @@ public function testInvalidBookingCantBeCreated()
     }
 
     /** @test */
-    public function testOverlappingBookings()
+    public function testOverlappingBookingsCannotBeCreatedOverlappingOnStart()
     {
-        $position = factory(Position::class)->create()->id;
-        $position2 = factory(Position::class)->create()->id;
+        $this->bypassRatingChecks();
 
         // Create two bookings for position 1
         factory(Booking::class)->create([
-            'position_id' => $position,
+            'position_id' => $this->position->id,
             'from' => '2019-08-10 14:00:00',
             'to' => '2019-08-10 15:00:00',
         ]);
 
-        $booking2 = factory(Booking::class)->create([
-            'position_id' => $position,
-            'from' => '2019-08-10 19:00:00',
-            'to' => '2019-08-10 20:00:00',
-        ]);
-
         // Test 1: Doesn't allow to book inside of booking
         $this->graphQL("
           mutation {
-            createBooking(user_id: 1300001, position: {
-                connect: {$position}
-            }, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
+            createBooking(user_id: {$this->mockUserId}, position_id: {$this->position->id}, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
                 id
             }
           }")->assertJsonPath('errors.0.message', "Can't have overlapping bookings for the same position!");
+    }
+
+    /** @test */
+    public function testOverlappingBookingsCannotBeCreatedOverlappingOnEnd()
+    {
+        $this->bypassRatingChecks();
+
+        factory(Booking::class)->create([
+            'position_id' => $this->position->id,
+            'from' => '2019-08-10 19:00:00',
+            'to' => '2019-08-10 20:00:00',
+        ]);
 
         $this->graphQL("
           mutation {
-            updateBooking(id: {$booking2->id}, from:\"2019-08-10 14:30:00\", to:\"2019-08-10 15:30:00\") {
+            createBooking(user_id: {$this->mockUserId}, position_id: {$this->position->id}, from:\"2019-08-10 18:30:00\", to:\"2019-08-10 19:30:00\") {
                 id
             }
           }")->assertJsonPath('errors.0.message', "Can't have overlapping bookings for the same position!");
+    }
+
+    /** @test */
+    public function testAllowsOverlappingBookingsOnDifferentPositions()
+    {
+        $this->bypassRatingChecks();
+
+        $creationJsonFormat = ['data' => ['createBooking' => ['id']]];
+        // twr position to meet the default rating check in setupOverlappingTest.
+        $differentPosition = factory(Position::class)->create(['callsign' => 'EGFF_TWR']);
 
```

Comment:
This test doesn't create a booking to check against?

Reply:
Two bookings are created in the graphql mutations.

Reply:
Yep fair enough